sympy/integrals/integrals.py

```diff
@@ -694,6 +694,14 @@ def eval_factored(f, x, a, b):
                         try:
                             evalued = Add(*others)._eval_interval(x, a, b)
                             evalued_pw = piecewise_fold(Add(*piecewises))._eval_interval(x, a, b)
+                            if isinstance(evalued_pw, Integral):
+                                antideriv = self._eval_integral(function, xab[0], manual=True)
+                                piecewises = []
+                                for f in Add.make_args(antideriv):
+                                    if any(isinstance(g, Piecewise)
+                                           for g in Mul.make_args(f)):
+                                               piecewises.append(piecewise_fold(f))
+                                evalued_pw = piecewise_fold(Add(*piecewises))._eval_interval(x, a, b)
```

Comment:
Here you recreate it for those that were Piecewise. What I do not understand is how do you know that `antideriv` from the manual integrator will end up with the same `Piecewise` sections as the original integral. That is, how can you replace evaluated_pw with this line?

It does indeed seem to work for the example, but the logic is not obvious to me.

Reply:
I mean I missed some part of the code here since i was not sure of this solution there will be needing of a isinstance check for type indef integrals and rest which are both not piecewise and indef integrals but that code is already above this so basically all i did was repeating some part of code but can we do something like making the below part of code a function and call it twice?https://github.com/sympy/sympy/blob/c39a6848728a3c84a92f348a4fc5f1a4f4ad2b84/sympy/integrals/integrals.py#L682-L693